packages/sdk/identity/src/offchain-data-wrapper.ts

```diff
@@ -173,12 +175,15 @@ class StorageRoot {
 
     const accounts = await this.wrapper.kit.contracts.getAccounts()
     if (await accounts.isAccount(this.account)) {
-      const signers = await Promise.all([
+      const keys = await Promise.all([
         accounts.getVoteSigner(this.account),
         accounts.getValidatorSigner(this.account),
         accounts.getAttestationSigner(this.account),
+        accounts.getDataEncryptionKey(this.account),
       ])
-      if (signers.some((signer) => signer === guessedSigner)) {
+      const signers = [keys[0], keys[1], keys[2], publicKeyToAddress(ensureUncompressed(keys[3]))]
```

Comment:
Can `getDataEncryptionKey()` return `undefined`? I see checks for it elsewhere in the code so we might want to add that check here before calling `ensureUncompressed()`